src/engine/execute.js

```diff
@@ -198,6 +227,9 @@ var execute = function (sequencer, thread) {
         getParam: function (paramName) {
             return thread.getParam(paramName);
         },
+        setProcedureReturn: function (value) {
+            thread.stackFrames[0].executionContext.return = value;
```

Comment:
That runs from the calling block and not the hat block so it has to "find" the stackFrame